OSDOCS-3939: [GCP] Enable user specified networking tags (#50178)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

> |Parameter|Description|Values

Author: I think this is due to the addition of some super long parameters like `platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKeyServiceAccount`. Is there a way to change the table's wrapping behavior? Otherwise we will need to rethink how we list these deeply nested parameters.

> |Parameter|Description|Values

Reviewer: I'm unaware of a simple way to do this with adoc. You could play with the various table and cell width options, though you'd need to verify that the table looks good on both the portal and d.o.c. :fearful:

|Either the default `pd-ssd` or the `pd-standard` disk type. The control plane nodes must be the `pd-ssd` disk type. Compute nodes can be either type.

|`platform.gcp.defaultMachinePlatform.tags`
|Optional. Additional network tags to add to the control plane and compute machines.
|One or more strings, for example "network-tag1".

|`platform.gcp.defaultMachinePlatform.type`
|The link:https://cloud.google.com/compute/docs/machine-types[GCP machine type] for control plane and compute machines.
|The GCP machine type, for example "n1-standard-4". 

|`platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKey.name`
|The name of the customer managed encryption key to be used for machine disk encryption.
|The encryption key name.

|`platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKey.keyRing`
|The name of the Key Management Service (KMS) key ring to which the KMS key belongs.
|The KMS key ring name.

|`platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKey.location`
|The link:https://cloud.google.com/kms/docs/locations[GCP location] in which the KMS key ring exists.
|The GCP location.

|`platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKey.projectID`

> |The ID of the project in which the KMS key ring exists. This value defaults to the value of `platform.gcp.projectID` if not set.

Reviewer: Could include a noun here as a boon to translation:

```suggestion
|The ID of the project in which the KMS key ring exists. This value defaults to the value of the `platform.gcp.projectID` property if it is not set.
```

Author: Good feedback. I chose "parameter" instead of "property".

|The GCP project ID.

> |`platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKey.kmsKeyServiceAccount`

Reviewer 2: It should be `platform.gcp.defaultMachinePlatform.osDisk.encryptionKey.kmsKeyServiceAccount` instead.

Author: Good catch, thanks.

> |The GCP Compute Engine System service account used for the encryption request for both control plane and compute machines. For more information on GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].

Reviewer 2: Suggest to replace `The GCP Compute Engine System service account used for the encryption request for both control plane and compute machines. ` with `KMSKeyServiceAccount is the service account being used for the encryption request for the given KMS key. If absent, the Compute Engine default service account is used. `.

Author: I would like to leave the mention of "control plane and compute machines" so that the user knows that this global value affects both sets of machines. I have added the sentence about the default service account.

|For control plane machines, the ID of the project in which the KMS key ring exists. This value defaults to the VM project ID if not set.
|The GCP project ID.

|`controlPlane.platform.gcp.osDisk.encryptionKey.kmsKeyServiceAccount`

> |The GCP Compute Engine System service account used for the encryption request for control plane machines. For more information on GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].

Reviewer 2: Suggest to replace `The GCP Compute Engine System service account used for the encryption request for control plane machines.` with `KMSKeyServiceAccount is the service account being used for the encryption request for the given KMS key. If absent, the Compute Engine default service account is used.`.

Author: Same change as above.

> |The GCP service account used for the encryption request for control plane machines. If absent, the Compute Engine default service account is used. For more information on GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].

Reviewer: "on" or "about" service accounts?

(reflects use throughout PR.)

Author: Switched them all over to "about", thanks

|The GCP service account email, for example `<service_account_name>@<project_id>.iam.gserviceaccount.com`.

|`controlPlane.platform.gcp.osDisk.diskSizeGB`
|The size of the disk in gigabytes (GB). This value applies to control plane machines.

> |Any size between 16 GB and 65536 GB.

Reviewer: Is this a string? Integer? Other?

Author: Reworded as "Any integer between 16 and 65536."

> |`compute.platform.gcp.osDisk.diskType`

Reviewer 2: I guess you mean `controlPlane.platform.gcp.osDisk.diskType` instead.

Author: Good catch, made a copy and paste error. Thank you

|The link:https://cloud.google.com/compute/docs/disks#disk-types[GCP disk type] for compute machines.

> |Compute machines must use the `pd-ssd` disk type, which is the default.

Reviewer 2: `Compute machines` or `Control plane machines`?

|`controlPlane.platform.gcp.tags`
|Optional. Additional network tags to add to the control plane machines. If set, this parameter overrides the `platform.gcp.defaultMachinePlatform.tags` parameter for control plane machines.

> |One or more strings, for example "control-plane-tag1".

Reviewer: The way we style "for example" throughout the install-config docs is inconsistent. That said, I prefer this way and will espouse it to all who might entertain it: 

```suggestion
|One or more strings. For example, "control-plane-tag1".
```

Obviously a less than advisory comment. :)

|`controlPlane.platform.gcp.type`
|The link:https://cloud.google.com/compute/docs/machine-types[GCP machine type] for control plane machines. If set, this parameter overrides the `platform.gcp.defaultMachinePlatform.type` parameter. 
|The GCP machine type, for example "n1-standard-4".

|`controlPlane.platform.gcp.zones`
|The availability zones where the installation program creates control plane machines.
|A list of valid link:https://cloud.google.com/compute/docs/regions-zones#available[GCP availability zones], such as `us-central1-a`, in a

> link:https://yaml.org/spec/1.2/spec.html#sequence//[YAML sequence].

Reviewer: Confused by the double slashes at the end of the link, but that seems to be the correct approach.

Author: Yeah, I tried playing around with that and it doesn't jump to the anchor without the trailing slashes. 🤷

|For compute machines, the ID of the project in which the KMS key ring exists. This value defaults to the VM project ID if not set.
|The GCP project ID.

|`compute.platform.gcp.osDisk.encryptionKey.kmsKeyServiceAccount`

> |The GCP Compute Engine System service account used for the encryption request for compute machines. For more information on GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].

Reviewer 2: Suggest to replace `The GCP Compute Engine System service account used for the encryption request for compute machines.` with `KMSKeyServiceAccount is the service account being used for the encryption request for the given KMS key. If absent, the Compute Engine default service account is used.`.

Author: Same change as above.

> |The GCP service account used for the encryption request for compute machines. If absent, the Compute Engine default service account is used. For more information on GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].

Reviewer: If the account is absent? If the value is blank? If the property is not included in the YAML?

Author: Reworded as "If this value is not set,"

|The GCP service account email, for example `<service_account_name>@<project_id>.iam.gserviceaccount.com`.

|`compute.platform.gcp.osDisk.diskSizeGB`
|The size of the disk in gigabytes (GB). This value applies to compute machines.

> |Any size between 16 GB and 65536 GB.

Reviewer: Type?

Author: Reworded as "Any integer between 16 and 65536."

> |`compute.platform.gcp.osDisk.diskType``

Reviewer 2: One additional back-quote in the end, please remove.

Reviewer 2: Besides, my screen which is 1920 x 1080 resolution cannot show all the 3 columns of the table, if without horizontal-scrolling. To be honest, I have to open 2 windows, one to show the first column of the table, another to show the other 2 columns. Is it possible to enhance? Thanks!

Author: Thanks, I fixed the typo. I don't think there is anything we can do to help with the table formatting. It might be easier to just view the source code rather than the rendered output. Sorry!

|The link:https://cloud.google.com/compute/docs/disks#disk-types[GCP disk type] for compute machines.
|Either the default `pd-ssd` or the `pd-standard` disk type.

|`compute.platform.gcp.tags`
|Optional. Additional network tags to add to the compute machines. If set, this parameter overrides the `platform.gcp.defaultMachinePlatform.tags` parameter for compute machines.

> |One or more strings, for example "compute-network-tag1".

Reviewer: ```suggestion
|One or more strings, for example `"compute-network-tag1"`.
```
Same comment for similar situations.

Author: With a string value that doesn't need to be in quotation marks in the configuration file, should I list it as `compute-network-tag` or `"compute-network-tag"`?

Reviewer: Oh, yeah, in that case, no quotes. :+1:

|String, for example, `/<datacenter_name>/vm/<folder_name>/<subfolder_name>`.

|`platform.vsphere.resourcePool`

> |Optional. The absolute path of an existing resource pool where the installation program creates the virtual machines. If you do not specify a value, the installation program will install the resources in the root of the cluster `/<datacenter_name>/host/<cluster_name>/Resources`.

Reviewer: Present tense rewrite as food for thought:

```suggestion
|Optional. The absolute path of an existing resource pool where the installation program creates the virtual machines. If you do not specify a value, the installation program installs the resources in the root of the cluster under `/<datacenter_name>/host/<cluster_name>/Resources`.
```

Author: Good catch, thanks
